### PR TITLE
Reenable the 'use auto property' lightbulb.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
A couple of changes have been made here.  We no longer implement this feature as a
'compilation end' analyzer.  Instead, it's a 'symbol analyzer'.  This should ideally
mean that the results it produces are available much quicker since we don't have to
wait for the compiltion analysis to end.

Previously this was a compilation analyzer because we wanted to also analyze all
usages to see if switching to an auto-prop would break things.  Now we eagerly
detect that you can convert to an auto prop, but we defer the conflict check
to when you try to apply the feature (like we do for "inline temp").

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ public bool SupportsFormattingOnTypedCharacter(Document document, char ch)
             var options = document.Project.Solution.Workspace.Options;
             var smartIndentOn = options.GetOption(FormattingOptions.SmartIndent, document.Project.Language) == FormattingOptions.IndentStyle.Smart;
 
-            if ((ch == '}' && !options.GetOption(FeatureOnOffOptions.AutoFormattingOnCloseBrace, document.Project.Language) && !smartIndentOn) ||
+                if ((ch == '}' && !options.GetOption(FeatureOnOffOptions.AutoFormattingOnCloseBrace, document.Project.Language) && !smartIndentOn) ||
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs]: Not sure how this file changed.  will revert.

**Comment by CyrusNajmabadi** [comment]
retest linux please

**Comment by CyrusNajmabadi** [comment]
retest unit please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest prtest/lin/dbg/unit32 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/win/dbg/unit32 please

**Comment by davkean** [comment]
tag @dotnet/roslyn-ide

**Pull request closed without merging**